### ISSUE TITLE
workspace/didChangeWatchedFiles registration

### ISSUE THREAD

**Issue by mattmassicotte** [castwide/solargraph#546]
Based on some source inspection of [vscode-solargraph](https://github.com/castwide/vscode-solargraph/blob/90def7ca555b103ea24794f14461a6e031076659/src/language-client.ts#L63), it looks like Solargraph wants to receive notifications of file changes. However, other servers I've worked with send a `client/registerCapability`, asking for that to be configured.

Should I just be firing off `workspace/didChangeWatchedFiles` without Solargraph registering, or is there something else my client should be looking for to know what patterns to watch for?

**Comment by castwide** [comment]
Solargraph responds to `workspace/didChangeWatchedFiles` messages regardless of whether the client registered the capability. In vscode-solargraph, the client decides which files to watch. It hasn't been a problem so far, although it does make sense for the server to decide which files to watch.

**Comment by mattmassicotte** [comment]
Yeah that makes sense. As of right now, Solargraph's file monitoring requires a custom-written client to work. The LSP spec needs the server to send that `client/registerCapability` message.
